cdc/owner/owner_state.go

```diff
@@ -0,0 +1,754 @@
+			zap.Uint64("checkpointTs", checkpointTs))
+	}
+
+	patch := &orchestrator.DataPatch{
+		Key: util.NewEtcdKey(kv.GetEtcdKeyChangeFeedStatus(cfID)),
+		Fun: func(old []byte) ([]byte, error) {
+			var changeFeedStatus model.ChangeFeedStatus
+
+			if old != nil {
+				err := json.Unmarshal(old, &changeFeedStatus)
+				if err != nil {
+					return nil, cerrors.ErrUnmarshalFailed.Wrap(err)
+				}
+
+				if changeFeedStatus.CheckpointTs > checkpointTs {
+					log.Panic("checkpointTs regressed",
+						zap.Reflect("changeFeedStatus", changeFeedStatus),
+						zap.Uint64("newCheckpointTs", checkpointTs))
+				}
+			}
+
+			changeFeedStatus.CheckpointTs = checkpointTs
+			changeFeedStatus.ResolvedTs = resolvedTs
+
+			newBytes, err := json.Marshal(&changeFeedStatus)
+			if err != nil {
+				return nil, cerrors.ErrMarshalFailed.Wrap(err).GenWithStackByArgs()
+			}
+
+			return newBytes, nil
+		},
+	}
+
+	s.patches = append(s.patches, patch)
+}
+
+// DispatchTable dispatches table to a given capture.
+func (s *ownerReactorState) DispatchTable(cfID model.ChangeFeedID, captureID model.CaptureID, tableID model.TableID, replicaInfo model.TableReplicaInfo) {
+	patch := &orchestrator.DataPatch{
+		Key: util.NewEtcdKey(kv.GetEtcdKeyTaskStatus(cfID, captureID)),
+		Fun: func(old []byte) ([]byte, error) {
+			var taskStatus model.TaskStatus
+			if len(old) > 0 {
+				err := json.Unmarshal(old, &taskStatus)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+			}
+
+			if _, ok := taskStatus.Operation[tableID]; ok {
+				log.Debug("already dispatched, ignore", zap.Int("tableID", int(tableID)))
+				return nil, cerrors.ErrEtcdIgnore.GenWithStackByArgs()
+			}
+
+			if taskStatus.Tables == nil {
+				taskStatus.Tables = make(map[model.TableID]*model.TableReplicaInfo)
+			}
+
+			taskStatus.Tables[tableID] = &replicaInfo
+			operation := &model.TableOperation{
+				Delete:     false,
+				BoundaryTs: replicaInfo.StartTs,
+				Done:       false,
+				Status:     model.OperDispatched,
+			}
+
+			if taskStatus.Operation == nil {
+				taskStatus.Operation = make(map[model.TableID]*model.TableOperation)
+			}
+			taskStatus.Operation[tableID] = operation
+
+			newValue, err := json.Marshal(&taskStatus)
+			if err != nil {
+				return nil, cerrors.ErrMarshalFailed.Wrap(err).GenWithStackByArgs()
+			}
+
+			return newValue, nil
+		},
+	}
+
+	s.patches = append(s.patches, patch)
+}
+
+// StartDeletingTable initiates a delete operation.
+// NOTE: callers need to guarantee that the table is running normally in the latest Etcd state snapshot.
+func (s *ownerReactorState) StartDeletingTable(cfID model.ChangeFeedID, captureID model.CaptureID, tableID model.TableID) {
+	captureTaskStatuses, ok := s.TaskStatuses[cfID]
+	if !ok {
+		log.Panic("owner bug: changeFeedState not found", zap.String("cfID", cfID))
+	}
+
+	_, ok = captureTaskStatuses[captureID]
+	if !ok {
+		log.Panic("owner bug: capture not found", zap.String("captureID", captureID))
+	}
```

Comment:
How does the owner state is guaranteed to be accurate in the new framework, taking this assert as an example, is there any strong constraint between `s.TaskStatuses` and the function call of `StartDeletingTable`.

Recommend to add document or comment to list all constraints, such as operation sequence, memory state guarantee.
And when we should panic when meeting unexpected state, and when we can ignore the unexpected state error.